exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/list/ListProof.java

```diff
@@ -29,4 +31,10 @@
    * @param visitor a visitor to apply to this node
    */
   void accept(ListProofVisitor visitor);
+
+  /**
+   * Returns tree node type.
+   * {@link NodeType}
+   */
+  NodeType getNodeType();
```

Comment:
I am not sure that this method belongs here in the public API. Do I understand it correctly that it basically returns the same as `Object#getClass()`, mirroring the hierarchy, and is only used to ask a question "Is it a hash node?"? Also, the enum contains some _validator-specific_ constants like `NodeType.NONE`.

Reply:
I think we can leave it as it is, since we want to refactor it to checked/unchecked and remove this class from public API.

Reply:
`ListProof` won't be hidden from the users because the structure of the proof must remain open. I'd suggest here one of the following:
- Keep this method, but
    1. Move the NodeType to ListProof and make it public
    1. Document its variants.
    1. Remove the type that is not appropriate for any implementation (NONE).
- Remove this method:
    1. Keep NodeType as is in the validator that uses it.
    1. (**or**) Replace it with instanceof.